commonv5/ts/sonarqube/Endpoint.ts

```diff
@@ -16,26 +18,58 @@ export interface EndpointData {
 }
 
 export default class Endpoint {
-  constructor(
-    public type: EndpointType,
-    private readonly data: EndpointData,
-  ) {}
+  public type: EndpointType;
 
-  public get auth() {
-    if (!this.data.token && this.data.password && this.data.password.length > 0) {
-      return { user: this.data.username, pass: this.data.password };
-    }
-    return { user: this.data.token || this.data.username };
-  }
+  private readonly data: EndpointData;
 
-  public get organization() {
-    return this.data.organization;
+  private readonly proxy: tl.ProxyConfiguration | null;
+
+  constructor(type: EndpointType, data: EndpointData) {
+    this.type = type;
+    this.data = data;
+    // Remove trailing slash at the end of the base url, if any
+    if (this.data) {
+      this.data.url = this.data?.url.replace(/\/$/, "");
+    }
+    this.proxy = tl.getHttpProxyConfiguration();
   }
 
   public get url() {
     return this.data.url;
   }
 
+  public get auth(): { username: string; password: string } {
```

Comment:
`password` could previously be undefined, that is not the case anymore, it will now be an empty string. That is because that is generally what libraries expect.

eg [axios](https://github.com/axios/axios/blob/7009715369a50740ba2ce00534012c1caf269ad2/index.d.ts#L116) 

If you are setting the auth header by hand, you'll want to use
`Basic username:` (so password as empty string).